std/xml.d

```diff
@@ -1059,9 +1059,10 @@ class Tag
                 munch(s,whitespace);
                 reqc(s,'=');
                 munch(s,whitespace);
-                reqc(s,'"');
-                string val = decode(munch(s,"^\""), DecodeMode.LOOSE);
-                reqc(s,'"');
+                char quote = requireOneOf(s,"'\"");
+                char[] notQuote = ['^', quote];
```

Comment:
You could still make this a `char[2]` to avoid an allocation. I don't know how important that is here, though. You would have to make it `notQuote[]` below then.